src/main/java/chess/board/Board.java

```diff
@@ -1,108 +1,76 @@
 package chess.board;
 
 import chess.piece.Color;
-import chess.piece.MovedPawn;
 import chess.piece.Piece;
+import chess.position.File;
 import chess.position.Position;
+import chess.position.UnitMovement;
+import chess.score.FileSquares;
+import chess.score.Score;
+import java.util.Arrays;
 import java.util.Collections;
+import java.util.HashMap;
+import java.util.List;
 import java.util.Map;
+import java.util.Map.Entry;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
 
 public class Board {
 
-    private final Map<Position, Piece> pieces;
+    private final Map<Position, Square> squares;
 
-    Board(Map<Position, Piece> pieces) {
-        this.pieces = pieces;
-    }
-
-    public void move(Position source, Position destination, Color currentTurnColor) {
-        validatePosition(source, destination);
-        validateNoPiecesBetween(source, destination);
-
-        Piece piece = pieces.get(source);
-        validateTurn(piece, currentTurnColor);
-        validateMovable(source, destination, piece);
-        replacePiece(source, destination, piece, currentTurnColor);
-    }
-
-    private void validatePosition(Position source, Position destination) {
-        validatePieceExistsAt(source);
-        validateAllyPieceNotOnDestination(source, destination);
-    }
-
-    private void validatePieceExistsAt(Position source) {
-        if (hasNoPieceOn(source)) {
-            throw new IllegalArgumentException("출발 칸에 기물이 없습니다.");
-        }
-    }
-
-    private void validateAllyPieceNotOnDestination(Position source, Position destination) {
-        Piece sourcePiece = pieces.get(source);
-        if (hasSameColorPieceOn(destination, sourcePiece)) {
-            throw new IllegalArgumentException("도착 칸에 자신의 기물이 있습니다.");
+    public Board(Map<Position, Piece> pieces) {
+        squares = new HashMap<>();
+        for (Entry<Position, Piece> entry : pieces.entrySet()) {
+            squares.put(entry.getKey(), new Square(entry.getValue()));
```

Comment:
아실수도 있을것 같지만, 소소한 Stream 팁 입니다.

```suggestion
        this.squares = pieces.entrySet()
                .stream()
                .collect(Collectors.toMap(
                        Entry::getKey,
                        entry -> new Square(entry.getValue()))
                );
```

Reply:
생성자에서는 `this`를 쓰는 것을 이번에는 잊었네요 🤦‍♂️, 스트림은 봐도봐도 참 매력적인 것 같습니다.

Reply:
반영하지 않았던 이유는 가독성이었지만, `Stream`을 사용해 `Board`에서 `HashMap` 구현체의 의존을 제거할 수 있고, `this` 키워드를 보다 명확하게 작성할 수 있어 반영합니다.